File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack=[]
        hashmap={"(":")","{":"}","[":"]"}
        for i in range(len(s)):
            if s[i] in hashmap:
                stack.append(s[i]) 
            else:
                if not stack or s[i]!=hashmap[stack[-1]]:
                    return False
                stack.pop()
            
        return not stack
        

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        HashMap<Character,Character> map=new HashMap<>();
        map.put('{','}');
        map.put('(',')');
        map.put('[',']');

        for(int i=0;i<s.length();i++){
            char currentChar=s.charAt(i);
            if(map.containsKey(s.charAt(i))){
                stack.push(currentChar);
            }else{
                if(stack.isEmpty() || map.get(stack.peek())!=currentChar){
                    return false;
                }
                stack.pop();
            }
        }
        return stack.isEmpty();
    }
}